doc/source/composites.rst

```diff
@@ -2,48 +2,405 @@
     >>> composite = compositor([local_scene[0.6],
     ...                         local_scene[0.8],
     ...                         local_scene[10.8]])
+
+One important thing to notice is that there is an internal difference
+between a composite and an image. A composite is defined as a special
+dataset which may have several bands (like ``R``, ``G`` and ``B``
+bands). However, the data isn't stretched, or clipped or gamma
+filtered until an image is generated.  To get an image out of the
+above composite::
+
     >>> from satpy.writers import to_image
     >>> img = to_image(composite)
     >>> img.invert([False, False, True])
     >>> img.stretch("linear")
     >>> img.gamma(1.7)
     >>> img.show()
 
+This part is called ``enhancement``, and is covered in more detail
+`enhancements <enhancements.html>`_ section.
+
+
+DifferenceCompositor
+--------------------
+
+:class:`DifferenceCompositor` calculates a difference of two datasets::
+
+    >>> from satpy.composites import DifferenceCompositor
+    >>> compositor = DifferenceCompositor("diffcomp")
+    >>> composite = compositor([local_scene[10.8], local_scene[12.0]])
+
+FillingCompositor
+-----------------
+
+:class:`FillingCompositor` fills the missing values in three datasets
+with the values of another dataset:::
+
+    >>> from satpy.composites import FillingCompositor
+    >>> compositor = FillingCompositor("fillcomp")
+    >>> filler = local_scene[0.6]
+    >>> data_with_holes_1 = local_scene['ch_a']
+    >>> data_with_holes_2 = local_scene['ch_b']
+    >>> data_with_holes_3 = local_scene['ch_c']
+    >>> composite = compositor([filler, data_with_holes_1, data_with_holes_2,
+    ...                         data_with_holes_3])
+
+PaletteCompositor
+------------------
+
+:class:`PaletteCompositor` creates a color version of a single channel
+categorical dataset using a colormap::
+
+    >>> from satpy.composites import PaletteCompositor
+    >>> compositor = PaletteCompositor("palcomp")
+    >>> composite = compositor([local_scene['cma'], local_scene['cma_pal']])
+
+The palette should have a single entry for all the (possible) values
+in the dataset mapping the value to an RGB triplet.  Typically the
+palette comes with the categorical (e.g. cloud mask) product that is
+being visualized.
+
+DayNightCompositor
+------------------
+
+:class:`DayNightCompositor` merges two different composites.  The
+first composite will be placed on the day-side of the scene, and the
+second one on the night side.  The transition from day to night is
+done by calculating solar zenith angle (SZA) weighed average of the
+two composites.  The SZA can optionally be given as third dataset, and
+if not given, the angles will be calculated.  Width of the blending
+zone can be defined when initializing the compositor (default values
+shown in the example below).
+
+    >>> from satpy.composites import DayNightCompositor
+    >>> compositor = DayNightCompositor("dnc", lim_low=85., lim_high=95.)
```

Comment:
This fails with:

```TypeError: __init__() got multiple values for argument 'lim_low'```

`DayNightCompositor.__init__` takes no positional arguments:

https://github.com/pytroll/satpy/blob/dff69ad5dc8373961111b4e1956f83e92a68618c/satpy/composites/__init__.py#L905

Reply:
This compositor should probably change to having `name` as the first positional argument to match the parent class. Thoughts @pnuu ?

Reply:
I guess so, I'll update.